python/ql/lib/semmle/python/frameworks/Bottle.qll

```diff
@@ -0,0 +1,116 @@
+  module BottleModule {
+    /** Provides models for the `bottle.response` module */
+    module Response {
+      /** Gets a reference to the `bottle.response` module. */
+      API::Node response() { result = bottle().getMember("response") }
+
+      /**
+       * A call to the `bottle.BaseResponse.set_header` or `bottle.BaseResponse.add_header` method.
+       *
+       * See https://bottlepy.org/docs/dev/api.html#bottle.BaseResponse.set_header
+       */
+      class BottleResponseHandlerSetHeaderCall extends Http::Server::ResponseHeaderWrite::Range,
+        DataFlow::MethodCallNode
+      {
+        BottleResponseHandlerSetHeaderCall() {
+          this = response().getMember(["set_header", "add_header"]).getACall()
+        }
+
+        override DataFlow::Node getNameArg() {
+          result in [this.getArg(0), this.getArgByName("name")]
+        }
+
+        override DataFlow::Node getValueArg() {
+          result in [this.getArg(1), this.getArgByName("value")]
+        }
+
+        override predicate nameAllowsNewline() { none() }
+
+        override predicate valueAllowsNewline() { none() }
+      }
+
+      /** Provides models for the `bottle.request` module */
+      module Request {
+        /** Gets a reference to the `bottle.request` module. */
+        API::Node request() { result = bottle().getMember("request") }
+
+        private class Request extends RemoteFlowSource::Range {
+          Request() { this = request().asSource() }
+
+          override string getSourceType() { result = "bottle.request" }
+        }
+
+        /**
+         * Taint propagation for `bottle.request`.
+         *
+         * See https://bottlepy.org/docs/dev/api.html#bottle.request
+         */
+        private class InstanceTaintSteps extends InstanceTaintStepsHelper {
+          InstanceTaintSteps() { this = "bottle.request" }
+
+          override DataFlow::Node getInstance() {
+            result = request().getAValueReachableFromSource()
+          }
+
+          override string getAttributeName() {
+            result in ["headers", "query", "forms", "params", "json", "url"]
+          }
+
+          override string getMethodName() { none() }
+
+          override string getAsyncMethodName() { none() }
+        }
+      }
+
+      /** Provides models for the `bottle.headers` module */
+      module Headers {
+        /** Gets a reference to the `bottle.headers` module. */
+        API::Node headers() { result = bottle().getMember("response").getMember("headers") }
+
+        /** A dict-like write to a response header. */
+        class HeaderWriteSubscript extends Http::Server::ResponseHeaderWrite::Range, DataFlow::Node {
+          API::Node name;
+          API::Node value;
+
+          HeaderWriteSubscript() {
+            exists(API::Node holder |
+              holder = headers() and
+              this = holder.asSource() and
+              value = holder.getSubscriptAt(name)
+            )
+          }
```

Comment:
This code looks like it will work for both reads and writes to `bottle.response.headers`? 

might want to just do the same as we do for flask/werkzeug: https://github.com/github/codeql/blob/d4a072818f451a281d9faddaf44fab3264d457fc/python/ql/lib/semmle/python/frameworks/Werkzeug.qll#L216-L239